### ISSUE TITLE
Able to cause kubelet to panic

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#6916]
v1.1.1-301-ge88b251 - I think this is because I'm using DOCKER_HOST=http://.... but docker is listening on unix.  Not sure.

```
E0129 22:26:33.453143    4015 util.go:82] Recovered from panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/util.go:76
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/util.go:64
/usr/lib/golang/src/runtime/asm_amd64.s:402
/usr/lib/golang/src/runtime/panic.go:387
/usr/lib/golang/src/runtime/panic.go:42
/usr/lib/golang/src/runtime/sigpanic_unix.go:26
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubelet/dockertools/manager.go:339
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubelet/dockertools/manager.go:2067
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubelet/dockertools/manager.go:425
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:3158
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1669
/usr/lib/golang/src/runtime/asm_amd64.s:401
/usr/lib/golang/src/runtime/panic.go:387
/usr/lib/golang/src/runtime/hashmap.go:385
```

**Comment by smarterclayton** [comment]
Hrm, I may have wedged this node:

```
E0129 22:28:17.131299    4061 event.go:188] Server rejected event '&api.Event{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"openshiftdev.local.142e0627a52ee9a4", GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, InvolvedObject:api.ObjectReference{Kind:"Node", Namespace:"", Name:"openshiftdev.local", UID:"openshiftdev.local", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"Starting", Message:"Starting kubelet.", Source:api.EventSource{Component:"kubelet", Host:"openshiftdev.local"}, FirstTimestamp:unversioned.Time{Time:time.Time{sec:63589703297, nsec:32055204, loc:(*time.Location)(0x5025640)}}, LastTimestamp:unversioned.Time{Time:time.Time{sec:63589703297, nsec:32055204, loc:(*time.Location)(0x5025640)}}, Count:1, Type:"Normal"}': 'Event "openshiftdev.local.142e0627a52ee9a4" is invalid: involvedObject.namespace: invalid value '', Details: not required for node' (will not retry!)
E0129 22:28:17.238147    4061 util.go:82] Recovered from panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/util.go:76
/data/src/github.com/openshift/origin/Godeps/_workspace/src/k8s.io/kubernetes/pkg/util/util.go:64
/usr/lib/golang/src/runtime/asm_amd64.s:402
/usr/lib/golang/src/runtime/panic.go:387
/usr/lib/golang/src/runtime/panic.go:42
/usr/lib/golang/src/runtime/sigpanic_unix.go:26
```

Happens even after I fix it.

**Comment by liggitt** [comment]
heh, possibly related to https://github.com/kubernetes/kubernetes/pull/20271 ?

**Comment by smarterclayton** [comment]
My daemon is also in a hosed state

**Comment by smarterclayton** [comment]
It was because the docker socket was a directory.  Going through and adding self checks on the daemon, going to drop the "allow docker not to exist" behavior for all in one.

**Comment by smarterclayton** [comment]
Fixed by #6920
